src/vsr/replica.zig

```diff
@@ -3051,65 +2726,120 @@ pub fn ReplicaType(
                 .commit = if (command == .do_view_change) self.commit_min else self.commit_max,
             };
 
-            // The DVC anchor: Within the log suffix following the anchor, we have additional
-            // guarantees about the state of the log headers which allow us to tolerate certain
-            // gaps (by locally guaranteeing that the gap does not hide a break).
-            // See Example 2/3 for more detail.
-            const op_dvc_anchor = std.math.max(
-                self.commit_min,
-                // +1: We can have a full pipeline, but not yet have performed any repair.
-                // In such a case, we want to send those pipeline_prepare_queue_max headers in the
-                // DVC, but not the preceding op (which may belong to a different chain).
-                // This satisfies the DVC invariant because the first op in the pipeline is
-                // "connected" to the canonical chain (via its "parent" checksum).
-                //
-                // For example, as a follower, we might have received pipeline_prepare_queue_max
-                // headers in the SV message, but not done any repair before the next view change.
-                1 + self.op -| constants.pipeline_prepare_queue_max,
+            stdx.copy_disjoint(
+                .exact,
+                Header,
+                std.mem.bytesAsSlice(Header, message.body()),
+                headers.constSlice(),
             );
+            message.header.set_checksum_body(message.body());
+            message.header.set_checksum();
 
-            // How many headers should we send in the DVC?
-            // - The DVC "The headers must connect to the cluster's committed ops." must be upheld.
-            // - The headers may contains gaps.
-            // - The headers may not contain breaks.
-            const op_suffix_min = op_min: {
-                var op: u64 = self.op;
+            return message.ref();
+        }
 
-                if (command == .start_view) {
-                    // The primary starting a new view has a pristine log suffix — no gaps or breaks.
-                    while (op > self.commit_min) : (op -= 1) {
+        fn create_view_change_headers(self: *const Self) vsr.ViewChangeHeaders.BoundedArray {
```

Comment:
This is essentially the same logic from before (in `create_view_change_message()`) with a few changes:

- The `Ensure that if we started a DVC before a crash` block is new.
- In the start-view section, require 1 fewer header, since `commit_min` does not necessarily exist in the journal. (When `commit_min == op_checkpoint`, and `op_checkpoint` is overwritten or corrupt).
- Split the start-view section since it is called by in more situations (e.g. followers updating their superblock headers). In these cases we just include has much chained suffix as possible — all we strictly need is the head op, so this is quite forgiving.
- Fix Retired backup DVC generation (it previously skipped headers that should be been included, but in a way that didn't violate correctness, so it accidentally worked anyway :facepalm:).